ledger/ledger-on-memory/src/app/scala/com/daml/ledger/on/memory/Main.scala

```diff
@@ -96,6 +114,19 @@ object Main {
               )
             )
         }
+      parser
+        .opt[Boolean]("pre-execute")
+        .optional()
+        .text(
+          "Whether to enable pre-execution (mutually exclusive with batching and will disable it).")
```

Comment:
As you can only have one of pre-execution or batch, please add a `parser.checkConfig` check to make sure only one is set and fail fast if both are.

Reply:
Done.